### ISSUE TITLE
Support setting html attributes in select option tags

### ISSUE THREAD

**Issue by ScottMaclure** [powmedia/backbone-forms#141]
Currently the _arrayToHtml function only checks for "val" and "label" properties, when rendering <option> elements.

I propose also supporting an "attributes" object that is rendered as attributes on the option element.

Example:

``` javascript
option: {
    "val" : "001",
    "label": "Product 001",
    "attributes" : {
        "data-url" : "http://some.ajax.url/for/data/001"
    }
}
```

Becomes:

``` html
<option value="001" data-url="http://some.ajax.url/for/data/001">Product 001</option>
```

This would provide flexibility for UI.

Edit: This would probably be best addressed as  a generic feature for all field types. E.g. I'd like to be able to do the same for a text input field as well.
